### ISSUE TITLE
Documentation: Indicate which types are from out of AsmResolver namespace

### ISSUE THREAD

**Issue by th3spis** [Washi1337/AsmResolver#372]
### Description

Thank you for your incredible work.

While learning how to use it, and considering that Im a newby reversing advanced .NET assemblys, I got a little confused with the documentation. So it would be nice to indicate when you're referencing a type or an element from other namespace/library. For example, in the entry for [dynamic-methods](https://github.com/Washi1337/AsmResolver/blob/development/docs/dotnet/dynamic-methods.rst) I think the `DynamicMethod dynamicMethod = ...` refers to the type/class from Assembly.Reflection. Please correct me if I'm wrong.

While it might seem obvious and immediate for experienced people, it can be confusing for learners trying to code a solution using this library and looking at the docs. I dont think this is the case, but for example using dnlib there was even some overlaping with names and visual studio required to specify. 

By specifying this, it would be more clear from the beginning if (and when) we need to Load an assembly using Reflection or AsmResolver and where to use each.

**Comment by Washi1337** [comment]
I agree, some of the documentation can indeed use some work in clarity. 

For what it is worth though, none of the examples in the documentation except for one section in the reference importing chapter require loading an assembly via `System.Reflection`. Dynamic methods can be created without it, and AsmResolver itself does not rely on reflection apart from a very few select use cases.
